src/core/lib/surface/server.h

```diff
@@ -377,12 +378,24 @@ class Server : public InternallyRefCounted<Server> {
     if (shutdown_refs_.fetch_sub(2, std::memory_order_acq_rel) == 2) {
       MutexLock lock(&mu_global_);
       MaybeFinishShutdown();
+      // The last request in-flight during shutdown is now complete.
+      if (shutdown_ready_ != nullptr) {
+        GPR_ASSERT(!shutdown_ready_->HasBeenNotified());
+        shutdown_ready_->Notify();
+      }
     }
   }
-  void ShutdownUnrefOnShutdownCall() ABSL_EXCLUSIVE_LOCKS_REQUIRED(mu_global_) {
+  // Returns a notification pointer to wait on if there are requests in-flight,
+  // or null.
+  absl::Notification* ShutdownUnrefOnShutdownCall()
```

Comment:
suggest `GRPC_MUST_USE_RESULT absl::Notification* ShutdownUnrefOnShutdownCall()` - since it's critical this return value not be ignored.

Reply:
Done.